README.md

```diff
@@ -6,3 +6,145 @@
+  "status": "finished",
+  "results": [
+    {
+      // True if the test passed
+      "passed": true,
+      // Name of the test
+      "testName": "agent request query params",
+      // Duration of the request in milliseconds
+      "requestDurationMs": 322
+    },
+    {
+      // False if the test failed
+      "passed": false,
+      // Name of the test
+      "testName": "ingress request headers",
+      // Duration of the request in milliseconds
+      "requestDurationMs": 23,
+      // Reason why the test failed
+      "reason": "Expected header 'X-Forwarded-For' to be undefined",
+      "meta": {
+        "error": {
+          // Error thrown by the test
+          // ...
+        },
+        // Array of requests that we received from proxy for this test
+        "requestsFromProxy": [
+          {
+            // URL from our app that proxy sent request to
+            "url": "<MOCK_APP_HOST>/worker/pxresult",
+            // Request headers
+            "headers": {},
+            // Request method
+            "method": "POST"
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+
+As an alternative, you can use our CLI client:
+```bash
+npm exec -y "git+https://github.com/fingerprintjs/dx-team-mock-for-proxy-integrations-e2e-tests.git" --
+# URL of the mock server 
+--api-url="<API_URL>" 
+# CDN endpoint in your proxy integration
+--host="https://mock-test-inter-568-mock-app-tests.cfi-fingerprint.com"
+# CDN endpoint in your proxy integration 
+--cdn-proxy-path="worker/pxdownload" 
+# Ingress endpoint in your proxy integration
+--ingress-proxy-path="worker/pxresult"
+# First part of ii parameter sent by our proxy integration: <trafficName>/<integrationVersion>/type
+--traffic-name="fingerprint-pro-akamai"
+# Second part of ii parameter sent by our proxy integration: <trafficName>/<integrationVersion>/type
+--integration-version="1.0.1-snapshot.0"
+```
+
+## Adding new tests
+
+Test cases can be found in [src/app/test/cases](src/app/test/cases) directory, and look like this:
+```ts
+import { assert, assertToBeTruthy } from '../../service/assert'
+import { TestCase } from '../../types/testCase'
+
+const testCase: TestCase = {
+  // Name of the tests
+  name: 'agent request preserve header and query',
+  // Test function. It receives `api` object that can be used to send requests to the proxy integration.
+  test: async (api) => {
+    const query = new URLSearchParams()
+    query.set('customQuery', '123')
+
+    // `requestFromProxy` is the request that we received from proxy integration.
+    const { requestFromProxy } = await api
+      // or .sendRequestToIngress which sends request to the ingress endpoint
+      // or .sendRequestToCacheEndpoint which sends request to the cache endpoint
+      .sendRequestToCdn(query, { headers: { 'X-Custom': '123' } })
+    const { ii, customQuery } = requestFromProxy.query
+
+    assertToBeTruthy(ii.toString())
+    assertToBeTruthy(customQuery.toString())
```

Comment:
Can you update this example here?

Reply:
What exactly needs to be updated?

Reply:
There is no ii.toString() in the original code, just ii as parameter

Reply:
Got it, done!